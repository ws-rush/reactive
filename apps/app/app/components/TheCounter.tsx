type Props = {
  readonly initial: number
}

export default function TheCounter({ initial }: Props) {
  const [count, setCount] = useState(initial)

  const inc = () => {
    setCount((previousCount) => previousCount + 1)
  }

  const dec = () => {
    setCount((previousCount) => previousCount - 1)
  }

  return (
    <div>
      {count}
      <button
        className="inc"
        data-testid="inc-button"
        onClick={inc}
        type="button"
      >
        +
      </button>
      <button
        className="dec"
        data-testid="dec-button"
        onClick={dec}
        type="button"
      >
        -
      </button>
    </div>
  )
}
