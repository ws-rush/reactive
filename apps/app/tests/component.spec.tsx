import TheCounter from '../app/components/TheCounter'
import { cleanup, fireEvent, render } from '@testing-library/react'
import React from 'react'
import { afterEach, describe, expect, it } from 'vitest'

describe('Counter component', () => {
  afterEach(() => {
    cleanup() // This ensures components are unmounted after each test
  })

  it('should render', () => {
    const { getByText } = render(<TheCounter initial={10} />)
    expect(getByText('10')).toBeTruthy()
    expect(getByText('10')).toMatchSnapshot()
  })

  it('should be interactive', () => {
    const { getByTestId, getByText } = render(<TheCounter initial={0} />)
    expect(getByText('0')).toBeTruthy()

    expect(getByTestId('inc-button')).toBeTruthy()
    expect(getByTestId('dec-button')).toBeTruthy()

    fireEvent.click(getByTestId('inc-button'))
    expect(getByText('1')).toBeTruthy()

    fireEvent.click(getByTestId('dec-button'))
    expect(getByText('0')).toBeTruthy()
  })
})
