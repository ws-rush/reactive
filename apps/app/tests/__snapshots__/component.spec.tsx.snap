// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter component > should render 1`] = `
<div>
  10
  <button
    class="inc"
    data-testid="inc-button"
    type="button"
  >
    +
  </button>
  <button
    class="dec"
    data-testid="dec-button"
    type="button"
  >
    -
  </button>
</div>
`;
